File: benchmarks/pandas/bench_update.py
import pandas as pd
import numpy as np
import json
import time

N = 100_000
data = list(range(N))
other_data = [i * 10 if i % 3 == 0 else None for i in range(N)]

s = pd.Series(data, dtype=float)
o = pd.Series(other_data, dtype=float)

# Warm-up
for _ in range(20):
    sc = s.copy()
    sc.update(o)

iterations = 200
start = time.perf_counter()
for _ in range(iterations):
    sc = s.copy()
    sc.update(o)
total_ms = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "update",
    "mean_ms": total_ms / iterations,
    "iterations": iterations,
    "total_ms": total_ms,
}))

File: benchmarks/tsb/bench_update.ts
import { Series, seriesUpdate } from "../../src/index.ts";

const N = 100_000;
const data = Float64Array.from({ length: N }, (_, i) => i);
const other = Float64Array.from({ length: N }, (_, i) => (i % 3 === 0 ? i * 10 : null as unknown as number));

const s = new Series({ data });
const o = new Series({ data: other });

// Warm-up
for (let i = 0; i < 20; i++) {
  seriesUpdate(s, o);
}

const iterations = 200;
const start = performance.now();
for (let i = 0; i < iterations; i++) {
  seriesUpdate(s, o);
}
const total_ms = performance.now() - start;

console.log(
  JSON.stringify({
    function: "update",
    mean_ms: total_ms / iterations,
    iterations,
    total_ms,
  }),
);
